crates/automations/src/server.rs

```diff
@@ -152,15 +153,12 @@ async fn ready_tasks_iter(
     semaphore: &Arc<tokio::sync::Semaphore>,
     task_types: &[i16],
 ) {
-    // Block until at least one permit is available.
-    if semaphore.available_permits() == 0 {
-        let _ = semaphore.clone().acquire_owned().await.unwrap();
-    }
-
     // Acquire all available permits, and then poll for up to that many tasks.
+    // If there's no currently available permits, then just acquire one.
```

Comment:
The downside of this change, is that it may be the case that _multiple_ permits become available before this async fn is polled again, and we would fail to pick up those multiple permits (and take only one).

Is this change required for testing use cases? It seems like it shouldn't be, so long as the test harness can control what permits are released to the semaphore?

Reply:
The problem for tests was that `available_permits()` could return 0 because the harness holds the only available permit before this is called.

But here `n_permits` should be the larger value of `available_permits()` and `1`, so we'll still try to acquire multiple permits if they're available.

Reply:
the scenario i was thinking of is:
 - no permits are available. We set `n_permits` to max(1) = 1.
 - we call acquire_many_owned(1)
 - the executor is busy, and a number of other tasks complete releasing their permits before we're polled again.
 - we're polled again and acquire a single permit, even though more are available.

Reply:
just to close the loop on this, I ended up reverting this change along with the refactor to have `TestHarness` call `poll_task` directly.